### DISCUSSION TITLE
RabbitMQ cluster in microk8s

### DISCUSSION THREAD

**Discussion by aleRizzolo** [rabbitmq/rabbitmq-server#5094] [Questions] [answered]
Hello guys, I'm trying to deploy a RabbitMQ cluster in microk8s. I use the following config:

```
apiVersion: rabbitmq.com/v1beta1

kind: RabbitmqCluster

metadata:

labels:

app: rabbitmq

name: cluster

spec:

replicas: 1

service:

type: ClusterIP

annotations:

name: rabbit-service

persistence:

storageClassName: local-storage

storage: 1Gi
```

And the I created a LocalStorage:

```
apiVersion: storage.k8s.io/v1

kind: StorageClass

metadata:

name: local-storage

provisioner: kubernetes.io/no-provisioner

volumeBindingMode: WaitForFirstConsumer

```
And a PV

```
apiVersion: v1

kind: PersistentVolume

metadata:

name: local-pv

spec:

capacity:

storage: 1Gi

accessModes:

- ReadWriteOnce

persistentVolumeReclaimPolicy: Retain

storageClassName: local-storage

local:

path: /data/volumes/rabbit

nodeAffinity:

required:

nodeSelectorTerms:

- matchExpressions:

- key: kubernetes.io/hostname

operator: In

values:

- k8s-dev
```

When I apply the config for the cluster, Pod is always in Pending state with this message:

`0/1 nodes are available: 1 node(s) didn't find available persistent volumes to bind. preemption: 0/1 nodes are available: 1 Preemption is not helpful for scheduling`

Anyone knows how to solve this?

**Comment by ChunyiLyu** [comment, marked as answer]
Hi @aleRizzolo, I've never used microk8s before, but when you create a RabbitmqCluster, the [Cluster Operator](https://github.com/rabbitmq/cluster-operator) creates a statefulSet which will create PersistentVolumeClaims. There is no need to preemptive create PV yourself. I would suggest: try to create a RabbitmqCluster again (no need to create a PV for it yourself), and if pod is still stuck at pending due to persistent volume error, you can double check the PersistentVolumeClaims by `kubectl get pvc` and the corresponding PVC will follow a name pattern of `persistence-RMQNAME-server-PODNUMBER`. If you have further questions, please also include the version of Cluster operator you're using.

**Reply by aleRizzolo** [reply]
Thanks for your reply.
I deleted the PV and re-deployed the cluster. It's stuck in "Pending" and after ans inspect I get `"0/1 nodes are available: 1 pod has unbound immediate PersistentVolumeClaims. preemption: 0/1 nodes are available: 1 Preemption is not helpful for scheduling"`
The PVC inspect is:
`Warning  ProvisioningFailed  10s (x3 over 36s)  persistentvolume-controller  storageclass.storage.k8s.io "local-storage" not found`

**Reply by ChunyiLyu** [reply]
The error message is descriptive. There is something wrong with the storage class you've configured and I suggest follow the error `Warning ProvisioningFailed 10s (x3 over 36s) persistentvolume-controller storageclass.storage.k8s.io "local-storage" not found` to debug. You should make sure that `local-storage` is an available storage class for example you can manually create a PVC yourself referencing `local-storage` as the storage class to test.

**Reply by aleRizzolo** [reply]
Hello. I just solved the issue. Thanks for your help.
I created a storage class and then a PV.
I deleted the StorageClass and the PV (with 5Gi) and the applied again everything
